src/net.cpp

```diff
@@ -2796,7 +2803,9 @@ void NetCleanup()
         {
             // Since we are quitting, disconnect abruptly from the node rather than finishing up our conversation
             // with it.
+            LOCK(pnode->cs_vRecvMsg);
             pnode->vRecvMsg.clear();
```

Comment:
lets put these in scopes so that we don't lock both cs_vRecvMsg and cs_vSend at the same time

Reply:
Agreed, will do.

Reply:
done